bakery/src/scripts/link_extras.py

```diff
@@ -10,62 +10,102 @@
 import requests
 
 
+def load_canonical_list(canonical_list):
+    with open(canonical_list) as canonical_file:
+        canonical_books = json.load(canonical_file)["canonical_books"]
+        canonical_ids = [book["uuid"] for book in canonical_books]
+
+    return canonical_ids
+
+
+def load_assembled_collection(input_dir):
+    """load assembled collection"""
+    assembled_collection = f"{input_dir}/collection.assembled.xhtml"
+    return etree.parse(assembled_collection)
+
+
+def find_legacy_id(node):
+    """find legacy module id"""
+    link = node.attrib["href"]
+    return link.lstrip("/contents/")
+
+
+def get_target_uuid(server, legacy_id):
+    """get target module uuid"""
+    req = requests.get(f"https://{server}/content/{legacy_id}")
+    req.raise_for_status()
+
+    return req.url.split("/")[-1]
+
+
+def get_containing_books(server, module_uuid):
+    """get list of books containing module"""
+    req = requests.get(f"https://{server}/extras/{module_uuid}")
+    req.raise_for_status()
+
+    content = req.json()
+
+    return {
+        book["ident_hash"].split("@")[0]: book["title"]
+        for book in content["books"]
+    }
+
+
+def match_canonical_book(canonical_ids, containing_books):
+    """match uuid in canonical book list"""
+    if len(containing_books) == 1:
+        [match] = containing_books
+        return match
```

Comment:
Kinda unimportant, but binding a new variable doesn't really help us or reduce any complexity for us here. Destructuring the dict into a list masks the fact that containing_books is a dict.
`return list(containing_books.keys())[0]` could be used. Or clearly `return containing_books[0]` could be used if this was simply a list (depending on how you respond to my above comment.